src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -12,94 +12,127 @@ def setUp(self):
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_b['cbsdId'] = device_b['userId']
+    grant_b['operationParam']['maxEirp'] = config['grant']['maxEirp']
+    # Step 6: Load one ppa in SAS Test Harness
+    ppa_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    # Step 7: Load device_c within the PPA in SAS Test Harness
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_c['installationParam']['latitude'] = config['device_c']['latitude']
+    device_c['installationParam']['longitude'] = config['device_c']['longitude']
+    # Step 7: Load grant request for device_c in SAS Test Harness
+    grant_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_c['cbsdId'] = device_c['userId']
+    grant_c['operationParam']['maxEirp'] = config['grant']['maxEirp']
+    # Load device_d within the PPA to SAS UUT
+    device_d = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_d.json')))
+    device_d['installationParam']['latitude'] = config['device_d']['latitude']
+    device_d['installationParam']['longitude'] = config['device_d']['longitude']
+    # Load grant request for device_d to SAS UUT
+    grant_d = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_d['cbsdId'] = device_d['userId']
+    grant_d['operationParam']['maxEirp'] = config['grant']['maxEirp']
+    config = {
+
+                'url': url,
+                'device_a': device_a,
+                'device_b': device_b,
+                'device_c': device_c,
+                'device_d': device_d,
+                'grant_a': grant_a,
+                'grant_b': grant_b,
+                'grant_c': grant_c,
+                'grant_d': grant_d,
+                'ppa_a': ppa_a,
+                'esc_sensor': esc_sensor,
+                'dump_config_cbsd_1': config['dump_config_cbsd_1'],
+                'dump_config_cbsd_2': config['dump_config_cbsd_2'],
+                'dump_config_ppa': config['dump_config_ppa'],
+                'dump_config_esc': config['dump_config_esc'],
+                'certificateHash': config['certificateHash'],
+                'ca_cart': config['ca_cart']
+                }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_FAD_2_default_config)
+  def test_WINNF_FT_S_FAD_2(self, config_filename):
+    config = loadConfig(config_filename)
+    url = config['url']
+    ca_cart = config['ca_cart']
+    device_a = config['device_a']
+    device_b = config['device_b']
+    device_c = config['device_c']
+    device_d = config['device_d']
+    grant_a = config['grant_a']
+    grant_b = config['grant_b']
+    grant_c = config['grant_c']
+    grant_d = config['grant_d']
+    ppa_a = config['ppa_a']
+    esc_sensor = config['esc_sensor']
+    # Start the server
+    self._sas_server = startServer(url, ca_cart)
```

Comment:
I don't think it's a good practice to change the class in a given test case. I think this may destroy the isolation of the test cases and make debugging quite hard.